hw4_1.ts

```diff
@@ -0,0 +1,33 @@
+interface Calculator {
+  add: (a: number, b: number) => number;
+  subtract: (a: number, b: number) => number;
+  multiply: (a: number, b: number) => number;
+  divide: (a: number, b: number) => number;
+}
+
+const calculator: Calculator = {
+  add: (a, b) => a + b,
+  subtract: (a, b) => a - b,
+  multiply: (a, b) => a * b,
+  divide: (a, b) => (b !== 0 ? a / b : NaN),
+};
+
+function calculate(
+  calculator: Calculator,
+  operation: string,
```

Comment:
By converting your list of operations to an Enum, you can improve your code's readability, maintainability, and overall efficiency. Don't miss out on the benefits of this best practice - make the switch today!